src/test/java/org/xbill/DNS/hosts/HostsFileParserTest.java

```diff
@@ -171,6 +171,17 @@ void testBigFileIsNotCompletelyCachedA() throws IOException {
     assertEquals(1, hostsFileParser.cacheSize());
   }
 
+  @Test
+  void testBigFileCompletelyCachedA() throws IOException {
+    System.setProperty("dnsjava.hostsfile.max_size", 1024 * 1024 * 1024 + "");
+    HostsFileParser hostsFileParser = generateLargeHostsFile("testBigFileCompletelyCachedA");
+    hostsFileParser
+      .getAddressForHost(Name.fromConstantString("localhost-10."), Type.A)
+      .orElseThrow(() -> new IllegalStateException("Host entry not found"));
+    assertEquals(1280, hostsFileParser.cacheSize());
+    System.clearProperty("dnsjava.hostsfile.max_size");
```

Comment:
If the assert fails, the property is never unset. Please wrap this all in a try/finally construct, like in this test case:

https://github.com/dnsjava/dnsjava/blob/a6371af00867d13e5b3dbee040f640e93420fb21/src/test/java/org/xbill/DNS/NioTcpClientTest.java#L42-L47